lib/ecto/schema.ex

```diff
@@ -252,7 +252,7 @@ defmodule Ecto.Schema do
   defined so structs and changeset functionalities are available.
   """
 
-  @type t :: %{__struct__: atom}
```

Comment:
struct()	%{__struct__: atom()}

Is listed in docs but that is incomplete, it should list the optional(atom)=>any as well, yeah?